src/com/palmergames/bukkit/towny/huds/MapHUD.java

```diff
@@ -0,0 +1,161 @@
+package com.palmergames.bukkit.towny.huds;
+
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.exceptions.NotRegisteredException;
+import com.palmergames.bukkit.towny.exceptions.TownyException;
+import com.palmergames.bukkit.towny.object.Nation;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.TownBlockType;
+import com.palmergames.bukkit.towny.object.TownyWorld;
+import com.palmergames.bukkit.towny.object.Translation;
+import com.palmergames.bukkit.towny.object.WorldCoord;
+import com.palmergames.bukkit.util.Colors;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.entity.Player;
+import org.bukkit.scoreboard.DisplaySlot;
+import org.bukkit.scoreboard.Objective;
+import org.bukkit.scoreboard.Scoreboard;
+
+public class MapHUD {
+	private static int lineWidth = 20, lineHeight = 10;
+	
+	public static void toggleOn(Player player) {
+		Scoreboard board = Bukkit.getScoreboardManager().getNewScoreboard();
+		Objective objective = board.registerNewObjective("MAP_HUD_OBJ", "", "");
+		objective.setDisplaySlot(DisplaySlot.SIDEBAR);
+		
+		int score = lineHeight + 2;
+		for (int i = 0; i < lineHeight; i++) {
+			board.registerNewTeam("mapTeam" + i).addEntry(ChatColor.values()[i].toString());
```

Comment:
Store `ChatColor.values()` in a variable. `values()` will allocate a new array every call.